modules/combined/test/tests/umat/gap_heat_transfer_umat.i

```diff
@@ -0,0 +1,170 @@
+[GlobalParams]
+  displacements = 'disp_x disp_y disp_z'
+  temperature = temp
+[]
+
+[Mesh]
+  file = gap_heat_transfer_mesh.e
+[]
+
+[Functions]
+  [disp]
+    type = PiecewiseLinear
+    x = '0 2.0'
+    y = '0 1.0'
+  []
+  [temp]
+    type = PiecewiseLinear
+    x = '0     1'
+    y = '273 2000'
+  []
+  [pressure_function]
+    type = PiecewiseLinear
+    x = '0     1'
+    y = '0 200'
+  []
+[]
+
+[Variables]
+  [disp_x]
+  []
+  [disp_y]
+  []
+  [disp_z]
+  []
+  [temp]
+    initial_condition = 273
+  []
+[]
+
+[ThermalContact]
+  [thermal_contact]
+    type = GapHeatTransfer
+    variable = temp
+    primary = 2
+    secondary = 3
+    emissivity_primary = 0
+    emissivity_secondary = 0
+  []
+[]
+
+[Modules/TensorMechanics/Master/All]
+  volumetric_locking_correction = true
+  strain = FINITE
+  generate_output = 'strain_yy stress_yy'
+[]
+
+[Kernels]
+  [heat]
+    type = HeatConduction
+    variable = temp
+  []
+[]
+
+[BCs]
+  [move_right]
+    type = FunctionDirichletBC
+    boundary = '3'
+    variable = disp_x
+    function = disp
+  []
+
+  [fixed_x]
+    type = DirichletBC
+    boundary = '1'
+    variable = disp_x
+    value = 0
+  []
+  [fixed_y]
+    type = DirichletBC
+    boundary = '1 2 4'
+    variable = disp_y
+    value = 0
+  []
+  [fixed_z]
+    type = DirichletBC
+    boundary = '1 2 3 4'
+    variable = disp_z
+    value = 0
+  []
+
+  [temp_bottom]
+    type = FunctionDirichletBC
+    boundary = 1
+    variable = temp
+    function = temp
+  []
+  [temp_top]
+    type = DirichletBC
+    boundary = 4
+    variable = temp
+    value = 100
+  []
+  [Pressure]
+    [example]
+      boundary = 3
+      function = pressure_function
+    []
+  []
+[]
+
+[Materials]
+  # 1. Active for umat calculation
+  [umat]
+    type = AbaqusUMATStress
+    constant_properties = '1.0e6 0.3'
+    plugin = '../../../../tensor_mechanics/test/plugins/elastic_temperature'
+    num_state_vars = 0
+    temperature = temp
+  []
+  #  2. Active for reference MOOSE computations
```

Comment:
Maybe worth noting where the block of materials active for case 2 ends.

Reply:
all of them, I thought that'd be clear